### PR TITLE
Adding uap10.1 configuration to System.Collections and System.Resources.ResourceManager

### LINKED ISSUE #11284: Move `List<T>` and `EqualityComparer/Comparer<T>` to CoreLib

### PR DESCRIPTION
**Pull request by SedarG**
Fixes #11284 
The primary motivation behind this change is to move `List<T>, Comparer<T>, EqualityComparer<T>` from System.Collections (for uap101aot) to System.Private.CoreLib. The change required updating System.Resource.ResourceManager too. 

@jkotas, @tarekgh , PTAL
/cc @dotnet/corert-contrib, @weshaggard , @ericstj

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @ramarag

File: src/System.Resources.ResourceManager/ref/System.Resources.ResourceManager.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <AssemblyVersion>4.0.0.0</AssemblyVersion>
     <OutputType>Library</OutputType>
     <NuGetTargetMoniker>.NETStandard,Version=v1.0</NuGetTargetMoniker>
+    <PackageTargetFramework Condition="'$(TargetGroup)' == ''">netstandard1.0;uap10.1</PackageTargetFramework>
```
Comment (SedarG) [inline src/System.Resources.ResourceManager/ref/System.Resources.ResourceManager.csproj]: ResourceManager is being built a bit strange today: Contract assembly is being built as 4.0.0.0 despite the implementation being built as 4.0.3.0. I tried to fix it but it causes validation errors. The things we harvest, the things that we prefer to use from live builds, how we define the SupportedFrameworks for target platforms all gets confused when I did that. I tried to manipulate the HarvestStablePackages with the tricks that I know but couldn't have it kneel down. So I'm leaving this weird issue as is, to attentions of @ericstj , @tarekgh , @weshaggard

File: src/System.Resources.ResourceManager/pkg/aot/System.Resources.ResourceManager.pkgproj [hunk 2]
```diff
@@ -5,5 +5,10 @@
     <PackageTargetRuntime>aot</PackageTargetRuntime>
     <PreventImplementationReference>true</PreventImplementationReference>
   </PropertyGroup>
+  <ItemGroup>
```
Comment (AlexGhiondea) [inline src/System.Resources.ResourceManager/pkg/aot/System.Resources.ResourceManager.pkgproj]: /cc @alexperovich who is working in this area as well.

File: src/System.Resources.ResourceManager/ref/System.Resources.ResourceManager.csproj (hunk 1 above)
Comment (tarekgh) [inline src/System.Resources.ResourceManager/ref/System.Resources.ResourceManager.csproj]: did you try to make this library compile for .Net Standard 1.7 and pump the version number of the contract and the the library. what exactly the validation error you are seeing?

---
In reply to: [82248278](https://github.com/dotnet/corefx/pull/12420#discussion_r82248278) [](ancestors = 82248278)

**Pull request merged**